### ISSUE TITLE
Weird images

### ISSUE THREAD

**Issue by pranali3215** [groupgets/pylepton#7]
I am using a FLIR Lepton Camera and a raspberry pi3 to generate heat maps.
In the beginning, the camera gives me the correct output, however, after some time, it gives me following kind of images:
![img_0](https://cloud.githubusercontent.com/assets/13495334/22148932/f7b8ccde-df36-11e6-8479-07368851e556.jpg)
![img_1](https://cloud.githubusercontent.com/assets/13495334/22149061/b3a04b84-df37-11e6-8054-269bbfc30b9e.jpg)
![img_4](https://cloud.githubusercontent.com/assets/13495334/22149064/b769dc3a-df37-11e6-882f-45a688d94035.jpg)

After I restart the pi, it starts giving me correct images but deteriorates after some time again.
Does anyone have any idea why this can be?
I have an idea that this could be about connections with pi3 as I previously used this with pi1 and it never gave me any such troubles.
Are there any changes in the way lepton camera is connected to pi3 as compared to pi1?

**Comment by pranali3215** [comment]
Update:

I thought it was a hardware issue, so I tried it again with a new camera and a new Breakout Board. However, it still threw me same images.

I installed LeptonModule and executed raspberry_video. I ran it for approximately half an hour, and I never got any striped images.

It seems there is a difference in how LeptonModule and pylepton reads images.
Any help?

**Comment by dirkstoop** [comment]
I had a similar issue, using pylepton on a Raspberry Pi Zero. My malformed images typically showed a big horizontal band that was black with some white parts. I researched what speeds the SPI port on the Pi Zero supports, and apparently it's at 25MHz and any whole fraction of that. The pylepton code seems to be configured at 18MHz so that shouldn't work.

In Lepton.py, look for this line:
`  SPEED = 18000000`
I changed it to
`  SPEED = 25000000`

and the issue became less very rare.

After that, I changed it to:
`  SPEED = 25000000 / 2`

And I haven't seen a bad frame since. Not entirely sure it's solved, but sure looks like it.
